### ISSUE TITLE
 No imported tomograms

### ISSUE THREAD

**Issue by jjdjesus** [3dem/relion#1187]
During the data import this problem arose:

Working... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   
           Wrote STAR files for 0 tilt-series.                   serialem.py:170

but the files seem to have been read fine (see below the error). (These same TS were processed in a previous version of relion 5 without any problems).
Apparently when the list of tomogram_ids updates, it goes blank. I solved the problem by commenting line 116 of the serialem.py script.

116    #tomogram_ids = matches

I don't know if this was the right way to solve the problem or if I need to import the data in another way.

**Environment:**
 - OS: Rocky Linux 8.9
 - RELION-5.0-beta-3-commit-12cf15

**Job options:**
 - Type of job: Import

relion_python_tomo_import SerialEM  --tilt-image-movie-pattern "../FramesCollected/*.tif" --mdoc-file-pattern "../mdoc/*.mdoc" --nominal-tilt-axis-angle 84.1 --nominal-pixel-size 0.84 --voltage 300 --spherical-aberration 2.7 --amplitude-contrast 0.1 --tilt-image-movie-pattern "../FramesCollected/*.tif" --optics-group-name "" --dose-per-tilt-image 2.42 --prefix B5 --invert-defocus-handedness  --output-directory Import/job126/  --pipeline_control Import/job126/

**Error message:**

[17:43:46] Started import_tilt_series job.                        serialem.py:76
           Creating output directory at Import/job126/tilt_series serialem.py:80
           Found 49 mdoc files and 2024 image files.              serialem.py:86
           Wrote tilt-series data STAR file                      serialem.py:147
           Import/job126/tilt_series.star                                       
           writing per tilt-series STAR files...                 serialem.py:150
Working... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   
           Wrote STAR files for 0 tilt-series.                   serialem.py:170

**Comment by noneed2cry-o** [comment]
I also had this problem while trying to import a tilt-series using the Relion5 pipeline. It appears to me that the program takes the file name from the .mdoc file and matches it to the names of the tilt image files. If the name matches it appends the name to an empty list and then sets the tomogram IDs based on this list. For those who cannot edit the code to run the job, I found a workaround by copying one of my image files and setting the name to match the .mdoc file. So, if my .mdoc file is “tiltseries1.mrc.mdoc”, I copied an image file and named it “tiltseries1.tif”. This way the program finds the name “tiltseries1” and appends it to the matches list resulting in the tomogram ID being “tiltseries1”. This shouldn’t affect downstream processing as it appears this code is just related to setting the tomogram ID name and the image file names that contribute to the tilt series are contained within the .mdoc file. However, I am only processing one tilt series at a time and have not tested this workaround for importing multiple tilt series in the same job.

**Comment by dahui-y** [comment]
Yeah. Meet the same problem. Does anyone have the method to solve the problem except for modifying the root name?

**Comment by Wherefores** [comment]
Similar issue as original poster. As others above have mentioned RELION requires the frames to have a name that matches the ImageFile name in the mdoc files. I've generated a python script (thank you AI) that will parse mdoc files and rename the frames to match the ImageFile names. It will also make backups of the original files so if something goes wrong you won't lose anything. Feel free to try it out, we've tested on a few datasets and it works nicely for us.

This works in batch.

The script assumes a specific folder architecture in your relion project directory:
RELION
>frames
>mdocs

[rewrite_mdocs_subframe_and_frameID.py](https://github.com/user-attachments/files/25217560/rewrite_mdocs_subframe_and_frameID.py)
